sklearn/neighbors/classification.py

```diff
@@ -383,10 +408,111 @@ def predict(self, X):
 
             y_pred[inliers, k] = classes_k.take(mode)
 
-        if outliers:
-            y_pred[outliers, :] = self.outlier_label
+            if outliers:
+                if self.outlier_label == 'most_frequent':
+                    prior = np.bincount(_y[:, k])
+                    frequent_label_index = prior.argmax()
+                    y_pred[outliers, k] = classes_k[frequent_label_index]
 
+        if outliers and isinstance(self.outlier_label, (numbers.Integral,
```

Comment:
I don't know why we should constrain this to numbers. I realise we've now reserved the names 'raise' and 'most_frequent' (which could be a small problem) but in general, any string in self.classes_ is allowed as a label. And if the chosen label is not in self.classes_, it needs to be added in fit. To me this appears to be a bug in the current logic...